qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+                                      'got %i expecting 1.' % len(ch_qubits))
+
+                ch_ = self._config.drive(ch_qubits[0])
+
+            elif isinstance(chan, MeasureChannel):
+                if len(ch_qubits) != 1:
+                    raise CalibrationError('Too many qubits for drive channel: '
+                                      'got %i expecting 1.' % len(ch_qubits))
+
+                ch_ = self._config.measure(ch_qubits[0])
+
+            elif isinstance(chan, ControlChannel):
+                chs_ = self._config.control(ch_qubits)
+
+                if len(chs_) != 1:
+                    raise CalibrationError('Ambiguous number of control channels for '
+                                      'qubits {} and {}.'.format(qubits, chan.name))
+
+                ch_ = chs_[0]
+
+            else:
+                chs = tuple(_.__name__ for _ in [DriveChannel, ControlChannel, MeasureChannel])
+                raise CalibrationError('Channel must be of type {}.'.format(chs))
+
+            return ch_.index
+        else:
+            return chan.index
+
+    def parameter_value(self, name: str, chan: PulseChannel, valid_only: bool = True,
+                        group: str = 'default',
+                        cutoff_date: datetime = None) -> Union[int, float, complex]:
+        """
+        Retrieve the value of a calibrated parameter from those stored.
+
+        Args:
+            name: The name of the parameter to get.
+            chan: The channel for which we want the value of the parameter.
+            valid_only: Use only parameters marked as valid.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date.
+                Parameters generated after the cutoff date will be ignored. If the
+                cutoff_date is None then all parameters are considered.
+
+        Returns:
+            value: The value of the parameter.
+
+        Raises:
+            CalibrationError: if there is no parameter value for the given parameter name
+                and pulse channel.
+        """
+        #pylint: disable = raise-missing-from
+        try:
+            if valid_only:
+                candidates = [p for p in self._params[name][chan] if p.valid]
+            else:
+                candidates = self._params[name][chan]
+
+            candidates = [_ for _ in candidates if _.group == group]
+
+            if cutoff_date:
+                candidates = [_ for _ in candidates if _ <= cutoff_date]
+
+            candidates.sort(key=lambda x: x.date_time)
+
+            return candidates[-1].value
+        except KeyError:
+            raise CalibrationError('No parameter value for %s and channel %s' % (name, chan.name))
+
+    def get_schedule(self, name: str, qubits: Tuple[int, ...],
```

Comment:
Instead of defining `free_params` (i.e. experimental class needs to search for parameter name, since naming is up to users), we can generate value dict and use new `pulse.call` function behavior in the builder to bind parameters on the fly from the parameter database.

Reply:
Can you elaborate on how this would work?